src/common/src/array/arrow/arrow_iceberg.rs

```diff
@@ -102,7 +104,7 @@ impl ToArrow for IcebergArrowConvert {
             DataType::Time => self.time_type_to_arrow(),
             DataType::Timestamp => self.timestamp_type_to_arrow(),
             DataType::Timestamptz => self.timestamptz_type_to_arrow(),
-            DataType::Interval => self.interval_type_to_arrow(),
+            DataType::Interval => self.varchar_type_to_arrow(),
```

Comment:
What are the pros and cons of doing it this way compared to overriding `<IcebergArrowConvert as ToArrow>::interval_type_to_arrow` to return `arrow_schema::DataType::Utf8`?

Reply:
I think there is no difference.

Reply:
Then I would prefer updating `interval_type_to_arrow`, to prevent from accidentally calling it. The current implementation means its functionality is broken and should never be called.

It also means keeping `interval_type_to_arrow` and `interval_to_arrow` consistent.